compiler/src/main/scala/org/scalajs/core/compiler/JSEncoding.scala

```diff
@@ -263,4 +265,19 @@ trait JSEncoding extends SubComponent { self: GenJSCode =>
     if (js.isKeyword(name) || name(0).isDigit || name(0) == '$')
       "$" + name
     else name
+
+
+  private def outerFieldSymSyffix(sym: Symbol): String = {
+    def allOutersInSym(acc: Set[Symbol], sym: Symbol): Set[Symbol] = {
+      val outersInSym = sym.info.members.filter(_.isOuterField).toSet
+      sym.parentSymbols.foldLeft(acc ++ outersInSym)(allOutersInSym)
+    }
+
+    val outers = allOutersInSym(Set.empty, sym.owner).toArray.sortBy(_.toString)
+    val index = outers.indexOf(sym)
+    assert(index >= 0)
+
+    if (index == 0) "f"
+    else "f$" + index
```

Comment:
Did you try the latest version I pushed or the simple one that was just adding.
```scala
else if (sym.isOuterField)
  sym.owner.name
```
The current version seems to have some issues.